File: src/easy/232-implement-queue-using-stacks/implementQueueUsingStacks.js
var MyQueue = function() {
  this.queue = [];

  this.push = function(x) {
    this.queue.push(x);
  }

  this.pop = function() {
    return this.queue.shift();
  }

  this.peek = function() {
    return this.queue[0];
  }

  this.empty = function() {
    return !this.queue.length;
  }
};

/** 
 * @param {number} x
 * @return {void}
 */
MyQueue.prototype.push = function(x) {
  this.push(x);
  return;
};

/**
 * @return {number}
 */
MyQueue.prototype.pop = function() {
  return this.pop();
};

/**
 * @return {number}
 */
MyQueue.prototype.peek = function() {
  return this.peek();
};

/**
 * @return {boolean}
 */
MyQueue.prototype.empty = function() {
  return this.empty();
};

File: src/easy/232-implement-queue-using-stacks/implement_queue_using_stacks.py
class MyQueue:
    def __init__(self):
        self.queue = []

    def push(self, x: int) -> None:
        self.queue.append(x)
        return None

    def pop(self) -> int:
        return self.queue.pop(0)

    def peek(self) -> int:
        return self.queue[0]

    def empty(self) -> bool:
        return len(self.queue) == 0
